crates/api_models/src/relay.rs

```diff
@@ -0,0 +1,103 @@
+pub use common_utils::types::MinorUnit;
+use serde::{Deserialize, Serialize};
+use utoipa::ToSchema;
+
+use crate::enums;
+
+#[derive(Debug, ToSchema, Clone, Deserialize, Serialize)]
+pub struct RelayRequest {
+    /// The identifier that is associated to a resource at the connector to which the relay request is being made
+    #[schema(example = "7256228702616471803954")]
+    pub connector_resource_id: String,
+    /// Identifier of the connector ( merchant connector account ) to which relay request is being made
+    #[schema(example = "mca_5apGeP94tMts6rg3U3kR", value_type = String)]
+    pub connector_id: common_utils::id_type::MerchantConnectorAccountId,
+    /// The type of relay request
+    #[serde(rename = "type")]
+    pub relay_type: RelayType,
+    /// The data that is associated with the relay request
+    pub data: Option<RelayData>,
+}
+
+#[derive(Debug, ToSchema, Clone, Deserialize, Serialize)]
+#[serde(rename_all = "snake_case")]
+pub enum RelayType {
+    /// The relay request is for a refund
+    Refund,
+}
+
+#[derive(Debug, ToSchema, Clone, Deserialize, Serialize)]
+#[serde(rename_all = "snake_case", untagged)]
```

Comment:
do not use untagged for deserialization. You can rely on the `relay_type` to infer this. Can be changed in the subsequent PRs as well